staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto

```diff
@@ -1514,3 +1519,56 @@ message ReopenContainerLogRequest {
 
 message ReopenContainerLogResponse{
 }
+
+// Common options used for checkpointing and restoring.
+message CheckpointRestoreOptions {
+    // Location of the checkpoint archive used for export/import
+    string archive_location = 1;
+    // Instead of using a local checkpoint archive use a OCI image
+    // to store the checkpoint.
+    bool archive_is_oci_image = 2;
+}
+
+message CheckpointContainerOptions {
+    CheckpointRestoreOptions common_options = 1;
+    // Leave the container running after checkpointing
+    bool leave_running = 2;
+}
+
+message CheckpointContainerRequest {
+    // ID of the container to be checkpointed.
+    string container_id = 1;
+    // Options for checkpointing.
+    CheckpointContainerOptions options = 2;
+}
+
+message CheckpointContainerResponse {
+}
+
+message RestoreContainerOptions {
+    CheckpointRestoreOptions common_options = 1;
+    // ID of the sandbox into which the container should be restored
+    // If this is not specified the container will be restored
+    // in the sandbox it was checkpointed from.
+    string pod_sandbox_id = 2;
+    // Restoring multiple copies of a checkpointed container may
+    // require a new name for the restored container.
+    string name = 3;
+    // Restoring a container in another pod might require changing
+    // Kubernetes specific mounts if the mount source depends on
+    // the pod ID.
+    // This tells the container engine to replace existing mount
+    // sources with a new source.
+    map<string, string> change_mounts = 4;
+}
+message RestoreContainerRequest {
+    // ID of the container to be restored.
+    string id = 1;
```

Comment:
The runtimes pick an ID for containers, so we don't need this. The id in the response should be enough.

Reply:
dropped